# Change directory

In [1]:
pwd

'/Users/chengdan'

In [3]:
cd /Users/chengdan/Desktop/Python_Project_Dan_Cheng

/Users/chengdan/Desktop/Python_Project_Dan_Cheng


In [4]:
pwd

'/Users/chengdan/Desktop/Python_Project_Dan_Cheng'

# Process original data

In [5]:
import pandas as pd

Read "CCDS.20160908_origin.txt" as csv file:

In [7]:
data0 = pd.read_csv("CCDS.20160908_origin.txt",sep="\t")http://localhost:8889/notebooks/Desktop/Python_Project_Dan_Cheng.ipynb#Check-the-type-of-data0

Check the type of data0:

In [8]:
type(data0)

pandas.core.frame.DataFrame

Check the shape of data0:

In [9]:
data0.shape

(34297, 11)

#This data frame contains 34297 rows and 11 columns

Check the firs 5 rows of data0:

In [10]:
data0.head()

#chromosome  nc_accession       gene  gene_id  ccds_id ccds_status  \
0           1   NC_000001.8  LINC00115    79854  CCDS1.1   Withdrawn   
1           1  NC_000001.11     SAMD11   148398  CCDS2.2      Public   
2           1  NC_000001.11      NOC2L    26155  CCDS3.1      Public   
3           1  NC_000001.11    PLEKHN1    84069  CCDS4.1      Public   
4           1  NC_000001.11       HES4    57801  CCDS5.1      Public   

  cds_strand cds_from  cds_to  \
0          -        -       -   
1          +   925941  944152   
2          -   944693  959239   
3          +   966531  974574   
4          -   999058  999972   

                                       cds_locations match_type  
0                                                  -       None  
1  [925941-926012, 930154-930335, 931038-931088, ...  Identical  
2  [944693-944799, 945056-945145, 945517-945652, ...  Identical  
3  [966531-966613, 966703-966802, 970276-970422, ...  Identical  
4  [999058-999431, 999525-999612, 999691-999786, ...  Identical

#Now I can see that the column labels including: "#chromosome", "nc_accession", "gene", "gene_id", "ccds_id",
"ccds_status", "cds_strand", "cds_from", "cds_to", "cds_locations", "match_type"

Check the composition of the "match_type" column of data0:

In [11]:
data0.iloc[:,-1].value_counts()

Identical    34031
None           219
Partial         47
Name: match_type, dtype: int64

#I only want to take the data whose "match_type" is "Identical"

Check the composition of the "ccds_status" column of data0:

In [12]:
data0.iloc[:,5].value_counts()

Public                                32534
Withdrawn                              1731
Under review, withdrawal                 15
Reviewed, update pending                  7
Withdrawn, inconsistent annotation        6
Under review, update                      4
Name: ccds_status, dtype: int64

#I only want to take the data whose "ccds_status" is "Public"

In [13]:
import numpy as np

Take the data with "match_type"=="Identical" and "ccds_status"=="Public", and save as data1:

In [26]:
data1 = np.logical_and(data0.iloc[:,5]=="Public",data0.iloc[:,-1]=="Identical")

Check the type of data1:

In [18]:
type(data1)

pandas.core.series.Series

#Now the type of the data is changed from dataframe to series

Check how many genes meet the requirements "match_type"=="Identical" and "ccds_status"=="Public":

In [27]:
data1.value_counts()

True     32533
False     1764
Name: ccds_status, dtype: int64

#There are 32533 genes meet the requirements "match_type"=="Identical" and "ccds_status"=="Public"

In [32]:
data2 = data0[data1,:]

TypeError: '([0        False
1         True
2         True
3         True
4         True
5         True
6         True
7         True
8         True
9         True
10        True
11        True
12        True
13        True
14        True
15        True
16       False
17       False
18        True
19        True
20        True
21        True
22       False
23        True
24        True
25        True
26        True
27        True
28       False
29        True
         ...  
34267     True
34268     True
34269     True
34270     True
34271     True
34272     True
34273     True
34274     True
34275     True
34276     True
34277     True
34278     True
34279     True
34280     True
34281     True
34282     True
34283     True
34284     True
34285     True
34286     True
34287     True
34288     True
34289     True
34290     True
34291     True
34292     True
34293     True
34294     True
34295     True
34296     True
Name: ccds_status, Length: 34297, dtype: bool], slice(None, None, None))' is an invalid key